{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Connect to bio.tools API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# \n",
    "import requests\n",
    "\n",
    "# a GET request\n",
    "response = requests.get('https://dev.bio.tools/api/tool/?collectionID=West-Life&format=json',\n",
    "    # could  \"Accept\": \"application/json;charset=UTF-8\"\n",
    ")\n",
    "assert 200 == response.status_code\n",
    "btservices = response.json()['list']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Convert to einfracentral format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'validated'"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import json\n",
    "\n",
    "# options, helpdesk, userManual, trainingInformation, \n",
    "\n",
    "for btservice in btservices:\n",
    "    eiservice = {\n",
    "        # required terms \n",
    "        'name': btservice['name'],\n",
    "        'url': btservice['homepage'],\n",
    "        'order': btservice['download'][0]['url'] if btservice['download'] else btservice['homepage'], \n",
    "        \"providers\": ['West-Life'],\n",
    "\n",
    "        # Required, but no good match in bio.tools\n",
    "        'symbol': 'https://about.west-life.eu/network/west-life/templates/westlife/images/west-life.png', \n",
    "        \"category\": \"Category-Software\", \"subcategory\": \"Subcategory-Tool\",\n",
    "        \"places\": [\"Place-WW\"],\n",
    "        # see below \"lifeCycleStatus\": \"LifeCycleStatus-Production\", \"trl\": \"TRL-9\",\n",
    "        \"serviceLevelAgreement\": \"https://auth.west-life.eu/aai/AUP.pdf\", # but see below\n",
    "        'funding' : \"EC funding \",\n",
    "\n",
    "        # terms that map fairly well\n",
    "        # could register all 'providers': btservice['collectionID'], # nearest approximation\n",
    "        'version': btservice['version'][0],\n",
    "        \"lastUpdate\": btservice['lastUpdate'],\n",
    "        'description': btservice['description'],\n",
    "        \"languages\": ['Language-'+btservice['language']] if btservice['language'] else [\"Language-en\"],\n",
    "        'tags': [ t['term'] for t in btservice['topic']],\n",
    "        \"termsOfUse\":  [btservice['license']] or [\"https://auth.west-life.eu/aai/AUP.pdf\"], # but see below\n",
    "\n",
    "    }\n",
    "    if btservice['function']:\n",
    "        for f in btservice['function']:\n",
    "            if f['note']:\n",
    "                tagline = f['note']\n",
    "                eiservice['tagline'] = tagline if len(tagline)<100 else tagline[0:97]+'...'\n",
    "                break;\n",
    "    if 'Mature'==btservice['maturity']:\n",
    "        eiservice[\"lifeCycleStatus\"] = \"LifeCycleStatus-Production\"\n",
    "        eiservice[\"trl\"]= \"TRL-9\"\n",
    "    elif 'Emerging'==btservice['maturity']:\n",
    "        eiservice[\"lifeCycleStatus\"] = \"LifeCycleStatus-Beta\"\n",
    "        eiservice[\"trl\"]= \"TRL-7\" # permissible levels are 9,8,7\n",
    "    else:\n",
    "        print('maturity', btservice['maturity'])\n",
    "        \n",
    "    for d  in btservice['documentation']:\n",
    "        if 'Citation instructions' == d['type']:\n",
    "            pass\n",
    "        elif 'Terms of Use' == d['type']:\n",
    "            eiservice['termsOfUse'] = d['url']\n",
    "            eiservice['serviceLevelAgreement'] = d['url']\n",
    "    for d  in btservice['credit']:\n",
    "        # unfortunately \"max length for 'funding' is 500 chars\"\n",
    "        #if  'Funding agency' == d['typeEntity']:\n",
    "            # eiservice['funding'] = eiservice['funding']+', '+d['name']\n",
    "        if  'Primary Contact' in d['typeRole']:\n",
    "            eiservice['helpdesk'] = d['url']\n",
    "\n",
    "    # validate it        \n",
    "    HEADERS = {'Content-Type': 'application/json', 'Accept': 'application/json'}\n",
    "    response = requests.post('http://beta.einfracentral.eu/api/service/validate', \n",
    "        data=json.dumps(eiservice), headers=HEADERS)\n",
    "\n",
    "    assert 200 == response.status_code, response.json()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'collectionID': ['BioExcel', 'West-Life', 'INSTRUCT'],\n",
       "  'accessibility': [],\n",
       "  'additionDate': '2016-08-11T13:31:51Z',\n",
       "  'owner': 'm.e.trellet@uu.nl',\n",
       "  'cost': None,\n",
       "  'download': [{'url': 'https://github.com/haddocking/disvis',\n",
       "    'note': None,\n",
       "    'version': None,\n",
       "    'type': 'Source code'}],\n",
       "  'validated': 0,\n",
       "  'availability': None,\n",
       "  'publication': [{'doi': '10.1093/bioinformatics/btv333',\n",
       "    'version': None,\n",
       "    'pmid': None,\n",
       "    'type': 'Primary',\n",
       "    'pmcid': None,\n",
       "    'metadata': {'title': 'DisVis: Quantifying and visualizing accessible interaction space of distance-restrained biomolecular complexes',\n",
       "     'abstract': '© The Author 2015. Summary: We present DisVis, a Python package and command line tool to calculate the reduced accessible interaction space of distance-restrained binary protein complexes, allowing for direct visualization and quantification of the information content of the distance restraints. The approach is general and can also be used as a knowledge-based distance energy term in FFT-based docking directly during the sampling stage.',\n",
       "     'citationCount': 3,\n",
       "     'authors': [{'name': 'Van Zundert G.C.P.'}, {'name': 'Bonvin A.M.J.J.'}],\n",
       "     'date': '2015-10-01T00:00:00Z',\n",
       "     'journal': 'Bioinformatics'}}],\n",
       "  'homepage_status': 0,\n",
       "  'maturity': 'Mature',\n",
       "  'biotoolsCURIE': 'biotools:disvis_web_portal',\n",
       "  'elixirPlatform': [],\n",
       "  'elixirNode': [],\n",
       "  'canonicalID': None,\n",
       "  'version': ['1.0'],\n",
       "  'homepage': 'http://milou.science.uu.nl/services/DISVIS',\n",
       "  'function': [{'note': 'Visualisation of interaction space between two molecules',\n",
       "    'input': [],\n",
       "    'operation': [{'term': 'Protein structure analysis',\n",
       "      'uri': 'http://edamontology.org/operation_2406'}],\n",
       "    'cmd': None,\n",
       "    'output': []}],\n",
       "  'lastUpdate': '2018-09-18T12:06:18Z',\n",
       "  'otherID': [],\n",
       "  'description': 'DisVis allows you to visualizes and quantify the information content of distance restraints between macromolecular complexes.\\nIt performs a full and systematic 6 dimensional search of the three translational and rotational degrees of freedom to determine the number of complexes consistent with the restraints. In addition, it outputs the percentage of restraints being violated and a density that represents the center-of-mass position of the scanning chain corresponding to the highest number of consistent restraints at every position in space.',\n",
       "  'toolType': ['Web application'],\n",
       "  'editPermission': {'type': 'private', 'authors': []},\n",
       "  'language': [],\n",
       "  'link': [{'url': 'https://github.com/haddocking/disvis',\n",
       "    'note': None,\n",
       "    'type': 'Repository'}],\n",
       "  'downtime': None,\n",
       "  'biotoolsID': 'disvis_web_portal',\n",
       "  'topic': [{'term': 'Protein interactions',\n",
       "    'uri': 'http://edamontology.org/topic_0128'}],\n",
       "  'name': 'DisVis web portal',\n",
       "  'license': None,\n",
       "  'documentation': [{'url': 'http://milou.science.uu.nl/cgi/services/DISVIS/disvis/',\n",
       "    'note': None,\n",
       "    'type': 'Citation instructions'},\n",
       "   {'url': 'http://milou.science.uu.nl/cgi/services/DISVIS/disvis/help',\n",
       "    'note': None,\n",
       "    'type': 'General'}],\n",
       "  'credit': [{'typeRole': ['Provider'],\n",
       "    'name': 'Utrecht University',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': [None],\n",
       "    'name': 'EGI (www.egi.eu)',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Consortium',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European FP7 e-Infrastructure grant, WeNMR, grant no. 261572',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO) , TOP-PUNT grant no. 718.015.001',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO), VICI grant no. 700.96.442',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, West-Life VRE, grant no. 675858',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, INDICO-DataCloud, grant no. 653549',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, EGI-Engage, grant no. 654142',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, BioExcel, grant no. 675728',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO), ECHO grant no.711.011.009',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Provider'],\n",
       "    'name': 'uu.nl',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Primary contact'],\n",
       "    'name': None,\n",
       "    'url': 'http://ask.bioexcel.eu/c/disvis',\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': 'disvis.server@gmail.com'}],\n",
       "  'operatingSystem': ['Linux', 'Mac']},\n",
       " {'collectionID': ['BioExcel', 'WestLife', 'INSTRUCT'],\n",
       "  'accessibility': [],\n",
       "  'additionDate': '2016-10-30T05:33:50Z',\n",
       "  'owner': 'm.e.trellet@uu.nl',\n",
       "  'cost': None,\n",
       "  'download': [{'url': 'https://github.com/haddocking/powerfit',\n",
       "    'note': None,\n",
       "    'version': None,\n",
       "    'type': 'Source code'}],\n",
       "  'validated': 0,\n",
       "  'availability': None,\n",
       "  'publication': [{'doi': '10.3934/biophy.2015.2.73',\n",
       "    'version': None,\n",
       "    'pmid': None,\n",
       "    'type': 'Primary',\n",
       "    'pmcid': None,\n",
       "    'metadata': {'title': 'Fast and sensitive rigid-body fitting into cryo-EM density maps with PowerFit',\n",
       "     'abstract': '© 2015, Alexandre M.J.J. Bonvin, et al. Cryo-EM is a rapidly developing method to investigate the three dimensional structure of large macromolecular complexes. In spite of all the advances in the field, the resolution of most cryo-EM density maps is too low for de novo model building. Therefore, the data are often complemented by fitting high-resolution subunits in the density to allow for an atomic interpretation. Typically, the first step in the modeling process is placing the subunits in the density as a rigid body. An objective method for automatic placement is full-exhaustive six dimensional cross correlation search between the model and the cryo-EM data, where the three translational and three rotational degrees of freedom are systematically sampled. In this article we present PowerFit, a Python package and program for fast and sensitive rigid body fitting. We introduce a novel, more sensitive scoring function, the core-weighted local cross correlation, and show how it can be calculated using FFTs for fast translational cross correlation scans. We further improved the search algorithm by using optimized rotational sets to reduce rotational redundancy and by limiting the cryo-EM data size through resampling and trimming the density. We demonstrate the superior scoring sensitivity of our scoring function on simulated data of the 80S D. melanogaster ribosome and on experimental data for four different cases. Through these advances, a fine-grained rotational search can now be performed within minutes on a CPU and seconds on a GPU. PowerFit is free software and can be downloaded from https://github.com/haddocking/powerfit.',\n",
       "     'citationCount': 6,\n",
       "     'authors': [{'name': 'van Zundert G.C.P.'}, {'name': 'Bonvin A.M.J.J.'}],\n",
       "     'date': '2015-01-01T00:00:00Z',\n",
       "     'journal': 'AIMS Biophysics'}}],\n",
       "  'homepage_status': 0,\n",
       "  'maturity': 'Mature',\n",
       "  'biotoolsCURIE': 'biotools:powerfit_web_portal',\n",
       "  'elixirPlatform': [],\n",
       "  'elixirNode': [],\n",
       "  'canonicalID': None,\n",
       "  'version': ['2.0'],\n",
       "  'homepage': 'http://milou.science.uu.nl/services/POWERFIT',\n",
       "  'function': [{'note': 'Fit an atomic model with a electron density map.',\n",
       "    'input': [],\n",
       "    'operation': [{'term': 'Protein structure analysis',\n",
       "      'uri': 'http://edamontology.org/operation_2406'}],\n",
       "    'cmd': None,\n",
       "    'output': []}],\n",
       "  'lastUpdate': '2018-09-18T12:06:18Z',\n",
       "  'otherID': [],\n",
       "  'description': 'PowerFit automatically fits high-resolution atomic structures into cryo-EM densities. \\nTo this end it performs a full-exhaustive 6-dimensional cross-correlation search between the atomic structure and the density. It takes as input an atomic structure in PDB-format and a cryo-EM density with its resolution; and outputs positions and rotations of the atomic structure corresponding to high correlation values. PowerFit uses the local cross-correlation function as its base score. The score can optionally be enhanced by a Laplace pre-filter and/or a core-weighted version to minimize overlapping densities from neighboring subunits.',\n",
       "  'toolType': ['Web application'],\n",
       "  'editPermission': {'type': 'private', 'authors': []},\n",
       "  'language': [],\n",
       "  'link': [{'url': 'https://github.com/haddocking/powerfit',\n",
       "    'note': None,\n",
       "    'type': 'Repository'}],\n",
       "  'downtime': None,\n",
       "  'biotoolsID': 'powerfit_web_portal',\n",
       "  'topic': [{'term': 'Protein structure analysis',\n",
       "    'uri': 'http://edamontology.org/topic_2814'}],\n",
       "  'name': 'PowerFit web portal',\n",
       "  'license': None,\n",
       "  'documentation': [{'url': 'http://milou.science.uu.nl/cgi/services/POWERFIT/powerfit/',\n",
       "    'note': None,\n",
       "    'type': 'Citation instructions'},\n",
       "   {'url': 'http://milou.science.uu.nl/cgi/services/POWERFIT/powerfit/help',\n",
       "    'note': None,\n",
       "    'type': 'General'}],\n",
       "  'credit': [{'typeRole': ['Provider'],\n",
       "    'name': 'Utrecht University',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': [None],\n",
       "    'name': 'EGI (www.egi.eu)',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Consortium',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European FP7 e-Infrastructure grant, WeNMR, grant no. 261572',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO) , TOP-PUNT grant no. 718.015.001',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO), VICI grant no. 700.96.442',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, West-Life VRE, grant no. 675858',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, INDICO-DataCloud, grant no. 653549',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, EGI-Engage, grant no. 654142',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, BioExcel, grant no. 675728',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO), ECHO grant no.711.011.009',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Provider'],\n",
       "    'name': 'http://uu.nl',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Primary contact'],\n",
       "    'name': None,\n",
       "    'url': 'http://ask.bioexcel.eu/c/powerfit',\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': 'powerfit.server@gmail.com'}],\n",
       "  'operatingSystem': ['Linux', 'Mac']},\n",
       " {'collectionID': ['BioExcel', 'West-Life', 'INSTRUCT'],\n",
       "  'accessibility': [],\n",
       "  'additionDate': '2016-08-08T12:19:01Z',\n",
       "  'owner': 'a.m.j.j.bonvin@uu.nl',\n",
       "  'cost': None,\n",
       "  'download': [{'url': 'http://www.bonvinlab.org/software/haddock2.2/download/',\n",
       "    'note': None,\n",
       "    'version': None,\n",
       "    'type': 'Source code'},\n",
       "   {'url': 'http://www.bonvinlab.org/software/haddock2.2/download/',\n",
       "    'note': None,\n",
       "    'version': None,\n",
       "    'type': 'Binaries'}],\n",
       "  'validated': 0,\n",
       "  'availability': None,\n",
       "  'publication': [{'doi': '10.1016/j.jmb.2015.09.014',\n",
       "    'version': None,\n",
       "    'pmid': None,\n",
       "    'type': 'Primary',\n",
       "    'pmcid': None,\n",
       "    'metadata': {'title': 'The HADDOCK2.2 Web Server: User-Friendly Integrative Modeling of Biomolecular Complexes',\n",
       "     'abstract': '© 2015 The Authors. The prediction of the quaternary structure of biomolecular macromolecules is of paramount importance for fundamental understanding of cellular processes and drug design. In the era of integrative structural biology, one way of increasing the accuracy of modeling methods used to predict the structure of biomolecular complexes is to include as much experimental or predictive information as possible in the process. This has been at the core of our information-driven docking approach HADDOCK. We present here the updated version 2.2 of the HADDOCK portal, which offers new features such as support for mixed molecule types, additional experimental restraints and improved protocols, all of this in a user-friendly interface. With well over 6000 registered users and 108,000 jobs served, an increasing fraction of which on grid resources, we hope that this timely upgrade will help the community to solve important biological questions and further advance the field. The HADDOCK2.2 Web server is freely accessible to non-profit users at http://haddock.science.uu.nl/services/HADDOCK2.2.',\n",
       "     'citationCount': 107,\n",
       "     'authors': [{'name': 'Van Zundert G.C.P.'},\n",
       "      {'name': 'Rodrigues J.P.G.L.M.'},\n",
       "      {'name': 'Trellet M.'},\n",
       "      {'name': 'Schmitz C.'},\n",
       "      {'name': 'Kastritis P.L.'},\n",
       "      {'name': 'Karaca E.'},\n",
       "      {'name': 'Melquiond A.S.J.'},\n",
       "      {'name': 'Van Dijk M.'},\n",
       "      {'name': 'De Vries S.J.'},\n",
       "      {'name': 'Bonvin A.M.J.J.'}],\n",
       "     'date': '2016-02-22T00:00:00Z',\n",
       "     'journal': 'Journal of Molecular Biology'}},\n",
       "   {'doi': '10.1038/nprot.2010.32',\n",
       "    'version': None,\n",
       "    'pmid': None,\n",
       "    'type': 'Other',\n",
       "    'pmcid': None,\n",
       "    'metadata': {'title': 'The HADDOCK web server for data-driven biomolecular docking',\n",
       "     'abstract': 'Computational docking is the prediction or modeling of the three-dimensional structure of a biomolecular complex, starting from the structures of the individual molecules in their free, unbound form. HADDOC K is a popular docking program that takes a datadriven approach to docking, with support for a wide range of experimental data. Here we present the HADDOC K web server protocol, facilitating the modeling of biomolecular complexes for a wide community. The main web interface is user-friendly, requiring only the structures of the individual components and a list of interacting residues as input. Additional web interfaces allow the more advanced user to exploit the full range of experimental data supported by HADDOC K and to customize the docking process. The HADDOC K server has access to the resources of a dedicated cluster and of the e-NMR GRID infrastructure. Therefore, a typical docking run takes only a few minutes to prepare and a few hours to complete. © 2010 Nature Publishing Group.',\n",
       "     'citationCount': 592,\n",
       "     'authors': [{'name': 'De Vries S.J.'},\n",
       "      {'name': 'Van Dijk M.'},\n",
       "      {'name': 'Bonvin A.M.J.J.'}],\n",
       "     'date': '2010-04-15T00:00:00Z',\n",
       "     'journal': 'Nature Protocols'}}],\n",
       "  'homepage_status': 0,\n",
       "  'maturity': 'Mature',\n",
       "  'biotoolsCURIE': 'biotools:haddock_web_portal',\n",
       "  'elixirPlatform': [],\n",
       "  'elixirNode': [],\n",
       "  'canonicalID': None,\n",
       "  'version': ['2.2'],\n",
       "  'homepage': 'http://haddock.science.uu.nl/services/HADDOCK2.2',\n",
       "  'function': [{'note': '3D prediction of biomolecular complexes',\n",
       "    'input': [],\n",
       "    'operation': [{'term': 'Protein quaternary structure prediction',\n",
       "      'uri': 'http://edamontology.org/operation_3350'}],\n",
       "    'cmd': None,\n",
       "    'output': []}],\n",
       "  'lastUpdate': '2018-09-18T12:06:18Z',\n",
       "  'otherID': [],\n",
       "  'description': 'HADDOCK (High Ambiguity Driven protein-protein DOCKing) is an information-driven flexible docking approach for the modeling of biomolecular complexes. HADDOCK distinguishes itself from ab-initio docking methods in the fact that it encodes information from identified or predicted protein interfaces in ambiguous interaction restraints (AIRs) to drive the docking process. HADDOCK can deal with a large class of modeling problems including protein-protein, protein-nucleic acids and protein-ligand complexes.\\nMore information about HADDOCK2.2 can be found on the HADDOCK2.2 website (http://bonvinlab.org/software/haddock2.2)',\n",
       "  'toolType': ['Web application'],\n",
       "  'editPermission': {'type': 'private', 'authors': []},\n",
       "  'language': [],\n",
       "  'link': [{'url': 'https://github.com/haddocking',\n",
       "    'note': None,\n",
       "    'type': 'Repository'}],\n",
       "  'downtime': None,\n",
       "  'biotoolsID': 'haddock_web_portal',\n",
       "  'topic': [{'term': 'Protein interactions',\n",
       "    'uri': 'http://edamontology.org/topic_0128'}],\n",
       "  'name': 'HADDOCK web portal',\n",
       "  'license': None,\n",
       "  'documentation': [{'url': 'http://www.bonvinlab.org/software/haddock2.2/download/',\n",
       "    'note': None,\n",
       "    'type': 'Terms of use'},\n",
       "   {'url': 'http://www.bonvinlab.org/software/haddock2.2/',\n",
       "    'note': None,\n",
       "    'type': 'Citation instructions'},\n",
       "   {'url': 'http://bonvinlab.org/software/haddock2.2/manual',\n",
       "    'note': None,\n",
       "    'type': 'General'}],\n",
       "  'credit': [{'typeRole': ['Provider'],\n",
       "    'name': 'Utrecht University',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': [None],\n",
       "    'name': 'EGI (www.egi.eu)',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Consortium',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European FP7 e-Infrastructure grant, WeNMR, grant no. 261572',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO) , TOP-PUNT grant no. 718.015.001',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO), VICI grant no. 700.96.442',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, West-Life VRE, grant no. 675858',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, INDICO-DataCloud, grant no. 653549',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, EGI-Engage, grant no. 654142',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'European H2020 e-Infrastructure grant, BioExcel, grant no. 675728',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'Netherlands Organization for Scientific Research (NWO), ECHO grant no.711.011.009',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Provider'],\n",
       "    'name': 'uu.nl',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Primary contact'],\n",
       "    'name': None,\n",
       "    'url': 'http://ask.bioexcel.eu',\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': 'haddock.support@gmail.com'}],\n",
       "  'operatingSystem': ['Linux', 'Mac']},\n",
       " {'collectionID': ['West-Life', 'Instruct'],\n",
       "  'accessibility': [],\n",
       "  'additionDate': '2016-10-09T08:16:19Z',\n",
       "  'owner': 'Instruct',\n",
       "  'cost': 'Free of charge',\n",
       "  'download': [],\n",
       "  'validated': 0,\n",
       "  'availability': None,\n",
       "  'publication': [],\n",
       "  'homepage_status': 0,\n",
       "  'maturity': 'Emerging',\n",
       "  'biotoolsCURIE': 'biotools:virtualfoldervm',\n",
       "  'elixirPlatform': [],\n",
       "  'elixirNode': [],\n",
       "  'canonicalID': None,\n",
       "  'version': ['16.06.2'],\n",
       "  'homepage': 'https://appdb.egi.eu/store/vappliance/d6.1.virtualfoldervm/',\n",
       "  'function': [{'note': 'H2020 Westlife WP6 - D6.1-VirtualFolderVM is based on Ubuntu 14.04 system and provides basic web interface available at port 80. It provides internal file repository with limited scratch disc space available within virtual machine and accessible via WEBDAV protocol at http://[vmdomain:80/webdav \\nThe internal file repository can be used by installed application to execute processing tasks etc. It provides integration of external file repositories specific to user/group. Currently user specific EUDAT (B2DROP) can be mounted and data processed.',\n",
       "    'input': [],\n",
       "    'operation': [{'term': 'Utility operation',\n",
       "      'uri': 'http://edamontology.org/operation_2409'}],\n",
       "    'cmd': None,\n",
       "    'output': []}],\n",
       "  'lastUpdate': '2018-09-18T12:06:18Z',\n",
       "  'otherID': [],\n",
       "  'description': 'H2020 Westlife WP6 - D6.1-VirtualFolderVM is based on Ubuntu 14.04 system and provides basic web interface available at port 80. It provides internal file repository with limited scratch disc space available within virtual machine and accessible via WEBDAV protocol at http://[vmdomain:80/webdav \\nThe internal file repository can be used by installed application to execute processing tasks etc. It provides integration of external file repositories specific to user/group. Currently user specific EUDAT (B2DROP) can be mounted and data processed.',\n",
       "  'toolType': ['Web application'],\n",
       "  'editPermission': {'type': 'private', 'authors': []},\n",
       "  'language': [],\n",
       "  'link': [{'url': 'https://github.com/h2020-westlife-eu/west-life-wp6/tree/master/wp6-virtualfolder',\n",
       "    'note': None,\n",
       "    'type': 'Repository'}],\n",
       "  'downtime': None,\n",
       "  'biotoolsID': 'virtualfoldervm',\n",
       "  'topic': [{'term': 'Biology', 'uri': 'http://edamontology.org/topic_3070'}],\n",
       "  'name': 'Virtual Folder for Structural Biology Projects',\n",
       "  'license': None,\n",
       "  'documentation': [{'url': 'https://github.com/h2020-westlife-eu/west-life-wp6/blob/master/license',\n",
       "    'note': None,\n",
       "    'type': 'Terms of use'},\n",
       "   {'url': 'http://internal-wiki.west-life.eu/w/index.php?title=d6.1',\n",
       "    'note': None,\n",
       "    'type': 'General'}],\n",
       "  'credit': [{'typeRole': ['Developer'],\n",
       "    'name': 'Tomas Kulhanek',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Provider'],\n",
       "    'name': 'STFC',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Institute',\n",
       "    'email': None},\n",
       "   {'typeRole': [None],\n",
       "    'name': 'Instruct',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Consortium',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Contributor'],\n",
       "    'name': 'West-Life Horizon 2020',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Funding agency',\n",
       "    'email': None},\n",
       "   {'typeRole': ['Developer'],\n",
       "    'name': 'Tomas Kulhanek',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': None,\n",
       "    'email': 'tomas.kulhanek@stfc.ac.uk'},\n",
       "   {'typeRole': ['Primary contact'],\n",
       "    'name': 'Tomas Kulhanek',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': 'tomas.kulhanek@stfc.ac.uk'}],\n",
       "  'operatingSystem': ['Linux', 'Windows', 'Mac']},\n",
       " {'collectionID': ['West-Life'],\n",
       "  'accessibility': [],\n",
       "  'additionDate': '2016-10-30T05:58:49Z',\n",
       "  'owner': 'rjoosten',\n",
       "  'cost': 'Free of charge (with restrictions)',\n",
       "  'download': [],\n",
       "  'validated': 0,\n",
       "  'availability': None,\n",
       "  'publication': [{'doi': None,\n",
       "    'version': None,\n",
       "    'pmid': '25075342',\n",
       "    'type': 'Primary',\n",
       "    'pmcid': None,\n",
       "    'metadata': None}],\n",
       "  'homepage_status': 0,\n",
       "  'maturity': 'Mature',\n",
       "  'biotoolsCURIE': 'biotools:pdb_redo_server',\n",
       "  'elixirPlatform': [],\n",
       "  'elixirNode': [],\n",
       "  'canonicalID': None,\n",
       "  'version': ['6.26'],\n",
       "  'homepage': 'http://xtal.nki.nl/pdb_redo',\n",
       "  'function': [{'note': None,\n",
       "    'input': [],\n",
       "    'operation': [{'term': 'Residue contact calculation',\n",
       "      'uri': 'http://edamontology.org/operation_2950'},\n",
       "     {'term': 'Protein modelling (backbone)',\n",
       "      'uri': 'http://edamontology.org/operation_0479'},\n",
       "     {'term': 'Molecular model refinement',\n",
       "      'uri': 'http://edamontology.org/operation_0322'},\n",
       "     {'term': 'Protein model validation',\n",
       "      'uri': 'http://edamontology.org/operation_0321'},\n",
       "     {'term': 'Protein modelling (side chains)',\n",
       "      'uri': 'http://edamontology.org/operation_0480'}],\n",
       "    'cmd': None,\n",
       "    'output': []}],\n",
       "  'lastUpdate': '2018-09-18T12:06:18Z',\n",
       "  'otherID': [],\n",
       "  'description': 'The PDB_REDO web server allows users to optimise their crystallographic structure model in an automated way. The model is refined and rebuilt in the context of the experimental data. The server returns a new structure model, new electron density maps, validation data, a script to visualise model changes in COOT, and a settings file for further model refinement in REFMAC.',\n",
       "  'toolType': ['Web application'],\n",
       "  'editPermission': {'type': 'private', 'authors': []},\n",
       "  'language': [],\n",
       "  'link': [],\n",
       "  'downtime': None,\n",
       "  'biotoolsID': 'pdb_redo_server',\n",
       "  'topic': [{'term': 'X-ray diffraction',\n",
       "    'uri': 'http://edamontology.org/topic_2828'},\n",
       "   {'term': 'Structure analysis',\n",
       "    'uri': 'http://edamontology.org/topic_0081'}],\n",
       "  'name': 'PDB_REDO server',\n",
       "  'license': None,\n",
       "  'documentation': [{'url': 'http://www.cmbi.ru.nl/pdb_redo/references.html',\n",
       "    'note': None,\n",
       "    'type': 'Citation instructions'}],\n",
       "  'credit': [{'typeRole': ['Support'],\n",
       "    'name': 'Robbie P. Joosten',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': None,\n",
       "    'email': 'r.joosten@nki.nl'},\n",
       "   {'typeRole': ['Primary contact'],\n",
       "    'name': 'Robbie P. Joosten',\n",
       "    'url': None,\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': 'r.joosten@nki.nl'}],\n",
       "  'operatingSystem': []},\n",
       " {'collectionID': ['West-Life'],\n",
       "  'accessibility': [],\n",
       "  'additionDate': '2016-11-02T14:30:00Z',\n",
       "  'owner': 'rjoosten',\n",
       "  'cost': 'Free of charge',\n",
       "  'download': [],\n",
       "  'validated': 0,\n",
       "  'availability': None,\n",
       "  'publication': [{'doi': None,\n",
       "    'version': None,\n",
       "    'pmid': '19395596',\n",
       "    'type': 'Primary',\n",
       "    'pmcid': None,\n",
       "    'metadata': {'title': 'ProteinCCD: Enabling the design of protein truncation constructs for expression and crystallization experiments',\n",
       "     'abstract': 'ProteinCCD (CCD for Crystallographic Construct Design) aims to facilitate a common practice in structural biology, namely the design of several truncation constructs of the protein under investigation, based on experimental data or on sequence analysis tools. ProteinCCD functions as a meta-server, available online at http://xtal.nki.nl/ccd, that collects information from prediction servers concerning secondary structure, disorder, coiled coils, transmembrane segments, domains and domain linkers. It then displays a condensed view of all results against the protein sequence. The user can study the output and choose interactively possible starts and ends for suitable protein constructs. Since the required input to ProteinCCD is the DNA and not the protein sequence, once the starts and ends of constructs are chosen, the software can automatically design the oligonucleotides needed for PCR amplification of all constructs. ProteinCCD outputs a comprehensive view of all constructs and all oligos needed for bookkeeping or for direct copy-paste ordering of the designed oligonucleotides.',\n",
       "     'citationCount': 29,\n",
       "     'authors': [{'name': 'Mooij W.T.M.'},\n",
       "      {'name': 'Mitsiki E.'},\n",
       "      {'name': 'Perrakis A.'}],\n",
       "     'date': '2009-08-04T00:00:00Z',\n",
       "     'journal': 'Nucleic Acids Research'}}],\n",
       "  'homepage_status': 0,\n",
       "  'maturity': 'Mature',\n",
       "  'biotoolsCURIE': 'biotools:ccd',\n",
       "  'elixirPlatform': [],\n",
       "  'elixirNode': [],\n",
       "  'canonicalID': None,\n",
       "  'version': ['2.0'],\n",
       "  'homepage': 'http://xtal.nki.nl/ccd',\n",
       "  'function': [{'note': None,\n",
       "    'input': [],\n",
       "    'operation': [{'term': 'Sequence analysis',\n",
       "      'uri': 'http://edamontology.org/operation_2403'},\n",
       "     {'term': 'Primer and probe design',\n",
       "      'uri': 'http://edamontology.org/operation_2419'}],\n",
       "    'cmd': None,\n",
       "    'output': []}],\n",
       "  'lastUpdate': '2018-09-18T12:06:18Z',\n",
       "  'otherID': [],\n",
       "  'description': 'The Protein Crystallisation Construct Designer (ProteinCCD) is a tool to help to choose promising constructs for protein expression and crystallisation. A specific feature of CCD is that since it starts from the DNA sequence, it keeps track of the protein-DNA relationship. Thus, although all the analysis and construct choice is being done in the protein level, CCD can be used to suggest primers for PCR amplification of the chosen constructs, since it also knows the DNA sequence.',\n",
       "  'toolType': ['Web application'],\n",
       "  'editPermission': {'type': 'private', 'authors': []},\n",
       "  'language': [],\n",
       "  'link': [],\n",
       "  'downtime': None,\n",
       "  'biotoolsID': 'ccd',\n",
       "  'topic': [{'term': 'X-ray diffraction',\n",
       "    'uri': 'http://edamontology.org/topic_2828'},\n",
       "   {'term': 'Protein expression', 'uri': 'http://edamontology.org/topic_0108'},\n",
       "   {'term': 'Protein properties',\n",
       "    'uri': 'http://edamontology.org/topic_0123'}],\n",
       "  'name': 'Protein Crystallisation Construct Designer',\n",
       "  'license': None,\n",
       "  'documentation': [{'url': 'https://xtal.nki.nl/ccd/credits',\n",
       "    'note': None,\n",
       "    'type': 'Citation instructions'},\n",
       "   {'url': 'https://xtal.nki.nl/ccd/help', 'note': None, 'type': 'General'}],\n",
       "  'credit': [{'typeRole': ['Support'],\n",
       "    'name': 'Anastassis Perrakis',\n",
       "    'url': 'http://xtal.nki.nl/ccd',\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': None,\n",
       "    'email': None},\n",
       "   {'typeRole': ['Primary contact'],\n",
       "    'name': 'Anastassis Perrakis',\n",
       "    'url': 'http://xtal.nki.nl/ccd',\n",
       "    'note': None,\n",
       "    'orcidid': None,\n",
       "    'typeEntity': 'Person',\n",
       "    'email': None}],\n",
       "  'operatingSystem': []}]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "btservices"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
